### DISCUSSION TITLE
Request for drive base with time

### DISCUSSION THREAD

**Discussion by kkhanmd** [pybricks/support#1455] [Ideas]
Just like there is drive base with mm (distance) can you please introduce drive base with time as well? I know there is a work -around like put drive base and then put a wait like time, however having it one line will make it so cleaner.

**Comment by laurensvalk** [comment]
What is the underlying goal? Is it truly to drive with time, or is it because you'd like to drive into a wall to align the robot, or something similar?

**Reply by kkhanmd** [reply]
Few reasons 1)square against a wall 2) to prevent stall as well, lets say if the distance can not be achieved, like the drive base is stuck, the rest of the program will still continue after that time is achieved.

**Reply by laurensvalk** [reply]
Thank you. This comes up from time to time, and I think we could add a useful method like that, which may not necessarily be drive for time.

So maybe something that aligns with a wall, and stops when you hit the wall; or wheels get stuck or; after some time, whichever comes first.

See also https://github.com/pybricks/support/issues/1414 where others made related suggestions. Feel free to chime in with suggestions.

**Comment by MrGibbage** [comment]
I also wanted something like this to be built in, but ultimately decided against it. We agree that it is cleaner and easier for the kids to understand if it is one line, so the answer is, "do it!". Write a base class and build all of the movement commands that you will need. DriveForDistance. DriveForTime. DriveUntilStalled. Curve. Etc. etc. We worked on our BaseRobot class over the summer, and we have plans to do the same again this summer. Most of our methods are just a couple of lines long, and really pretty easy to understand. It's a great way for the more advanced kids to get their first introduction to Object-oriented programming. We did a lot of testing, and we documented everything the Judges loved that type of engineering process.

Good luck!

**Reply by kkhanmd** [reply]
How do I learn object oriented programming? I have zero programming knowledge, how do I lead my kids to do it? My kids are already spending hours and hours in innovation project and perfecting missions, I am scared they will not be willing to put even more hours to learn something that I don't know how to teach.

**Reply by MrGibbage** [reply]
In that case, that would be quite challenging. Not saying it is impossible though. Python is a great language to learn Object-oriented programing, and many people now get their first introduction to programming with python. There are many, many resources for python online. You could learn it along with your team. Just start with google. 

Not a perfect solution, and I wish I had more suggestions for you.
